XRefs to RefMan improvements

FabienLelaquais opened the pull request:
Addressed #434, #678, #694 and #697.

- Allow for duplicate API elements
  (in different packages)
- Rewrite dubious links to constructors
   (recommand the "\`Class,\_\_init\_\_^\`(\`Class\` constructor)" syntax) 
- Fixed some broken links

Commit:
- Allow for duplicate API elements
  (in different packages)
- Rewrite dubious links to constructors
- Fixed some broken links

Commit:
Debug trace removed

jrobinAV reviewed (approved):
I love it. Non need to think about adding every single method exposed at taipy level as an exception in the refman step !!

Awesome!

Commit:
- Add GUI tutorial's notebook file
- MkDocs structure fixed for run/deploy section
- Fix typos, conventions, ... in GUI tutorial sections

jrobinAV reviewed (approved):
Some good cleaning!

Commit:
- Set proper doc branch in notebook files
